src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java

```diff
@@ -55,6 +61,140 @@
+                        "INDEX_OP", "EXPR", "BLOCK_COMMENT_BEGIN", "PACKAGE_DEF",
+                        "IMPLEMENTS_CLAUSE", "NUM_FLOAT", "LITERAL_DO", "EOF", "GE", "RESOURCES",
+                        "MOD", "DEC", "EQUAL", "LITERAL_BOOLEAN", "CLASS_DEF", "COLON",
+                        "LITERAL_TRY", "ENUM_DEF", "GT", "NUM_INT", "ANNOTATION_DEF",
+                        "LITERAL_NATIVE", "METHOD_REF", "TYPE_ARGUMENTS", "DOUBLE_COLON", "IDENT",
+                        "MOD_ASSIGN", "LITERAL_FOR", "SUPER_CTOR_CALL", "STRING_LITERAL",
+                        "ARRAY_DECLARATOR", "LITERAL_CASE").collect(Collectors.toSet()));
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("EmptyBlock",
+                Stream.of("LITERAL_CATCH", "ARRAY_INIT", "LITERAL_DEFAULT", "LITERAL_CASE")
+                        .collect(Collectors.toSet()));
+        // too many false javadocs missing on override methods and
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("WriteTag",
+                Stream.of("ENUM_CONSTANT_DEF", "METHOD_DEF", "CTOR_DEF", "ANNOTATION_FIELD_DEF")
+                        .collect(Collectors.toSet()));
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("AnnotationLocation",
+                Stream.of("TYPECAST", "DOT", "TYPE_ARGUMENT", "ANNOTATION_DEF", "LITERAL_NEW",
+                        "LITERAL_THROWS", "PARAMETER_DEF", "IMPLEMENTS_CLAUSE",
+                        "ANNOTATION_FIELD_DEF").collect(Collectors.toSet()));
+        // not normally possible to keep these on 1 line due to line length restriction
+        TOKENS_IN_CONFIG_TO_IGNORE.put("NoLineWrap",
+                Stream.of("METHOD_DEF", "CTOR_DEF", "CLASS_DEF", "ENUM_DEF", "INTERFACE_DEF")
+                        .collect(Collectors.toSet()));
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("RightCurly",
+                Stream.of("METHOD_DEF", "CTOR_DEF", "CLASS_DEF", "INSTANCE_INIT", "LITERAL_FOR",
+                        "STATIC_INIT", "LITERAL_WHILE", "LITERAL_DO").collect(Collectors.toSet()));
+        // whitespace after is preferred
+        TOKENS_IN_CONFIG_TO_IGNORE.put("NoWhitespaceAfter",
+                Stream.of("TYPECAST", "LITERAL_SYNCHRONIZED").collect(Collectors.toSet()));
+        // some need context to decide if should be separated or not, others are
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("SeparatorWrap",
+                Stream.of("RBRACK", "AT", "ELLIPSIS", "SEMI", "LPAREN", "RPAREN",
+                        "ARRAY_DECLARATOR").collect(Collectors.toSet()));
+        // prefer no braces, lambda can be on or off depending on context
+        TOKENS_IN_CONFIG_TO_IGNORE.put("NeedBraces",
+                Stream.of("LITERAL_DEFAULT", "LITERAL_CASE", "LAMBDA").collect(Collectors.toSet()));
```

Comment:
LAMBDA - allowSingleLineStatement = true , all other should have braces

LITERAL_DEFAULT, LITERAL_CASE - braces in case/default of switches is good to avoid sharing scope of variables, but looks unusual, so we prefer to have no braces.